packages/strapi-plugin-users-permissions/services/Providers.js

```diff
@@ -109,6 +109,40 @@ const getProfile = async (provider, query, callback) => {
   }).get();
 
   switch (provider) {
+    case 'discord': {
+      const discord = new Purest({
+        provider: 'discord',
+        config: {
+          'discord': {
+            'https://discordapp.com/api/': {
+              '__domain': {
+                'auth': {
+                  'auth': {'bearer': '[0]'}
+                }
+              },
+              '{endpoint}': {
+                '__path': {
+                  'alias': '__default'
+                }
+              }
+            }
+          }
+        }
+      });
+      discord.query().get('users/@me').auth(access_token).request((err, res, body) => {
+        if (err) {
+          callback(err);
+        } else {
+          // Combine username and discriminator because discord username is not unique
+          var username = body.username + '#' + body.discriminator;
```

Comment:
A typical Discord "username" looks something like `DMehaffy#1337` hence the addition of the `#`